### PR TITLE
Allow < and > in chat and name

### PR DESCRIPTION
**Pull request by MacaylaMarvelous81**

### CODE REVIEW

File: serveProduction.js [hunk 1]
```diff
@@ -197,9 +197,8 @@ const processChat = async (channel_id, msg) => {
 
     const sanitizedChat = sanitizeChat(msg)
 
-    const request = "http://www.purgomalum.com/service/json?text=" + sanitizedChat
-    const playerNameRequest = "http://www.purgomalum.com/service/json?text=" + decodedMario.getPlayername()
-
+    const request = "http://www.purgomalum.com/service/json?text=" + encodeURIComponent(sanitizedChat)
+    const playerNameRequest = "http://www.purgomalum.com/service/json?text=" + encodeURIComponent(decodedMario.getPlayername())
```
Comment (MacaylaMarvelous81) [inline serveProduction.js:201]: Also start encoding the URI for the request.

Comment (MacaylaMarvelous81) [reply serveProduction.js:201]: Because & is used for seperating query params in URI. It should be encoded

Comment (github10omar) [reply serveProduction.js:201]: also allow rainbow mario. but this is a good thig tho, would like this in base game.

Comment (MacaylaMarvelous81) [reply serveProduction.js:201]: What

**Comment by Cooliokid956** [comment]
You seem to be new and have no idea why this was put in place.
It’s a long story, but people found a vulnerability in the game that allowed you to do stuff on other’s computers, so the made this. What you’re trying to do defeats the purpose of this.
Are you shitass disguised?

**Comment by TERRAERROR** [comment]
> You seem to be new and have no idea why this was put in place.
> It’s a long story, but people found a vulnerability in the game that allowed you to do stuff on other’s computers, so the made this. What you’re trying to do defeats the purpose of this.
> Are you shitass disguised?

this

**Comment by MacaylaMarvelous81** [comment]
Oh oops

**Comment by MacaylaMarvelous81** [comment]
It's encoded &lt &gt though

**Comment by Cooliokid956** [comment]
Why

**Comment by MacaylaMarvelous81** [comment]
The XSS prevented with &lt; and &gt;

**Comment by TERRAERROR** [comment]
still think this will still be a vulnerability, give me a bit to test this

**Comment by TERRAERROR** [comment]
seems like it's still running stuff if you do `<script>` (tried in game on my own server), looks like it still just unpatches xss.

**Comment by MacaylaMarvelous81** [comment]
It does not run the script for me [it looks like this](https://youtu.be/upT_bdce-vs)

**Comment by luphoria** [comment]
The way you are solving this error seems very insecure, and as @TERRAERROR is saying there is a good chance it doesn't even work. What if the requests to the external page don't work? What if it randomly updates? What if the owner is malicious? Forcing our project to run off of some random external sanitization is not a good idea... though we should fix the blocking of < and >, as when I implemented it it was very intended as a quick and easy fix to be updated in the coming days but it's been more than a month if I remember correctly.

**Comment by luphoria** [comment]
The script in the video is not running because you are not using async. Try a cross-site scripting method such as img onerror.

**Comment by dekrain** [comment]
There's something called `document.createTextNode`, so you can use this for chat

**Comment by luphoria** [comment]
@dekrain emotes would then not work

**Comment by dekrain** [comment]
Maybe split the message into text and emotes, and then sanitize the text parts with `document.createTextNode`

**Comment by sm64js** [comment]
Thank you for this, I will merge this in manually and test it

**Comment by sm64js** [comment]
I tried this out, it seems to me that I am still able to insert html tags that get recognized and processes as such. So I don't want to include this yet. I am not too concerned with the ability to use greater than and less than symbols in chat.

If I am mistaken and this does work and there is no vulnerability can you send a new PR updated to the commit and I will try again.

**Pull request closed without merging**